### DISCUSSION TITLE
数据库服务端部署时对接阿里云oss服务上传文件出错，求解决！！！

### DISCUSSION THREAD

**Discussion by TanXiang7o** [lobehub/lobehub#3590] [Self Hosting]
![image](https://github.com/user-attachments/assets/9fe8bffe-ddc1-433d-8261-095a9f568ab4)
![image](https://github.com/user-attachments/assets/23cfeb38-e923-4783-a66f-74fa8867549a)
S3_PUBLIC_DOMAIN=https://s3.oss-us-west-1.aliyuncs.com/lobechat-tx
S3_ENDPOINT=https://s3.oss-us-west-1.aliyuncs.com
S3_BUCKET=lobechat-tx
S3_SECRET_ACCESS_KEY=*
S3_ACCESS_KEY_ID=*
oss我本地已经测试过了是能正常上传访问的，不知道是什么问题

**Comment by arvinxx** [comment]
大概率是你配置问题，自己检查下？我没配过阿里云的 OSS，所以不知道你哪里配的有问题

**Reply by TanXiang7o** [reply]
我本地测的没问题啊，哥，你看我这几个环境变量哪个看着不对劲哦

**Reply by arvinxx** [reply]
大概率是 endpoint 不对

**Reply by TanXiang7o** [reply]
我本地测的这个是对的啊，阿里云的文档跟s3相关的只有这些了
![image](https://github.com/user-attachments/assets/1d1d786e-615b-4b82-ad42-557ed2fc4ab6)
![image](https://github.com/user-attachments/assets/c327e36c-ec8e-4197-9c9e-b93604f3641d)
我的bucket是地域属性的（美国硅谷），这个endpoint我本地也测过是没问题的
![image](https://github.com/user-attachments/assets/a29c14ad-e466-4fad-9492-93163069e312)
主要代码出错是在这里
![image](https://github.com/user-attachments/assets/449174e8-061a-48d6-8850-daf7ce639f27)
这个input.pathname是啥？
实在不行，我看能不能本地部署调试一下

**Reply by TanXiang7o** [reply]
![image](https://github.com/user-attachments/assets/21fafcd0-294a-44db-8200-6e25a095065a)
![image](https://github.com/user-attachments/assets/10503231-b252-4eb9-b321-bd7fb8437ef7)
endpoint应该是没问题的，我前面加了`s3.`，本地也都测过

**Comment by Sun-drenched** [comment]
S3_PUBLIC_DOMAIN配得不对吧，应该对应阿里云OSS控制台给的bucket域名

**Reply by arvinxx** [reply]
S3_PUBLIC_DOMAIN 影响的是上传后返回的地址，他这个是上传都上传不了，问题应该是出在 ENDPOINT 上

**Reply by Sun-drenched** [reply]
> S3_PUBLIC_DOMAIN 影响的是上传后返回的地址，他这个是上传都上传不了，问题应该是出在 ENDPOINT 上

他不是说endpoint测过没问题。。。不知道他为什么要在endpoint里面加个S3、S3_PUBLIC_DOMAIN也跟控制台给的对不上。
反正我用的阿里云OSS，控制台给的域名前加个https://填入变量就能用了。

**Reply by TanXiang7o** [reply]
你是部署在vercel上的吗

**Reply by TanXiang7o** [reply]
抽象了，是这个没删，其它的没问题
![image](https://github.com/user-attachments/assets/f603673e-9ccb-455c-91d8-db7e69ecc22d)
